### PR TITLE
Add a "Force update" button to PageSpeed Insights.

### PR DESCRIPTION
**Pull request by tofumatt**
## Summary

Addresses issue #87.

## Relevant technical choices

This changes the `getReport` resolver to always run and the checks to be based on selector/resolver status rather than if there's any data in the reducer, so that invalidated resolvers and an API cache clear like in the PR will cause PageSpeed Inisghts to refresh.

## Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/modules/pagespeed-insights/components/common/LabReportMetrics.js [hunk 1]
```diff
@@ -25,18 +25,44 @@ import classnames from 'classnames';
...
 
 export default function LabReportMetrics( { data, error } ) {
+	const referenceURL = useSelect( ( select ) => select( CORE_SITE ).getCurrentReferenceURL() );
+
+	const { invalidateResolution } = useDispatch( STORE_NAME );
+	const updateReport = useCallback( async ( event ) => {
+		event.preventDefault();
+
+		// Invalidate the PageSpeed API request caches.
+		await API.invalidateCache( 'modules', 'pagespeed' );
```
Comment (aaemnnosttv) [inline assets/js/modules/pagespeed-insights/components/common/LabReportMetrics.js:59]: The second parameter here should be the module slug. `pagespeed` is actually the datapoint name for the report selector, which we want to keep. See the control for `getReport` in `assets/js/modules/pagespeed-insights/datastore/report.js`
```suggestion
		await API.invalidateCache( 'modules', 'pagespeed-insights', 'pagespeed' );
```

File: assets/js/modules/pagespeed-insights/components/common/LabReportMetrics.js [hunk 2]
```diff
@@ -103,6 +129,12 @@ export default function LabReportMetrics( { data, error } ) {
 			</table>
 			<div className="googlesitekit-pagespeed-report__row googlesitekit-pagespeed-report__row--last">
 				<ReportDetailsLink />
+				<Button
+					href="#update"
```
Comment (aaemnnosttv) [inline assets/js/modules/pagespeed-insights/components/common/LabReportMetrics.js:133]: Why the `href` here? If we just want a normal `<button>` that won't submit, we can use `type="button"`.

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
Looks good, just a small fix and one question. I also noticed that the storybook stories for the component need an update or two as well.

Kapture 2020-09-23 at 11 44 50

Compare with the [storybook for the current release](https://google.github.io/site-kit-wp/storybook/master/?path=/story/pagespeed-insights-module-components--dashboard-widget-loading) and you'll see what I mean. Note the loading state is not rendering properly in both, so bonus points if you can fix that easily 😄

File: assets/js/modules/pagespeed-insights/components/common/LabReportMetrics.js (hunk 2 above)
Comment (tofumatt) [reply assets/js/modules/pagespeed-insights/components/common/LabReportMetrics.js:133]: Argh, that was a mistake, thanks! Didn't mean to commit it; non-`href` should be fine 👍🏻

**Comment by tofumatt** [comment]
Storybook is all fixed, including the loading screen, so this should be set now. Thanks!

**Review by aaemnnosttv — COMMENTED** [review]
Thanks @tofumatt – this looks good although there are a few details here which are not consistent with the AC. Particularly these:

- add a "Run test again" link in the bottom left corner
- "View details at PageSpeed Insights" is moved to the bottom right corner

The component currently looks like this on the lab tab:
image

The most obvious thing there is that the position is swapped; the button is said to be on the left but it is on the right.

Using some devtools magic, we can see what this would look like below

image

I think this looks better and is more consistent with other components by having the main action left-aligned. This may require other changes though for the details link to stay right aligned when there is no button rendered on the left (maybe an empty div is needed here to preserve the flex?).

Also "Run test again" is implemented as a button but the AC describes it as a link. I think this is more consistent with our other components as well as most actions are tied to buttons, although perhaps you could make the case that this is a "secondary" action so it shouldn't be a button.

Let's update the button positioning/alignment of the button and details link as well as double check with @felixarntz about the element type we should use for the action.

**Comment by felixarntz** [comment]
@tofumatt @aaemnnosttv 
> Also "Run test again" is implemented as a button but the AC describes it as a link. I think this is more consistent with our other components as well as most actions are tied to buttons, although perhaps you could make the case that this is a "secondary" action so it shouldn't be a button.

Yes let's make this visually appear like a link. It shouldn't look like a button since it's not a primary action on the screen where it appears. Our buttons that actually look like buttons are intended for primary actions.

**Comment by tofumatt** [comment]
Sounds good; updated and ready for another look! 👍🏻

File: assets/js/modules/pagespeed-insights/components/common/LabReportMetrics.js [hunk 3]
```diff
@@ -128,10 +128,10 @@ export default function LabReportMetrics( { data, error } ) {
 				</tbody>
 			</table>
 			<div className="googlesitekit-pagespeed-report__row googlesitekit-pagespeed-report__row--last">
-				<ReportDetailsLink />
-				<Button onClick={ updateReport }>
+				<Link href="#update-pagespeed" onClick={ updateReport }>
```
Comment (aaemnnosttv) [inline assets/js/modules/pagespeed-insights/components/common/LabReportMetrics.js:131]: The `href` shouldn't be needed here. It should render as a `button` that looks like a link without it. Can we avoid adding this?

**Review by aaemnnosttv — COMMENTED** [review]
Looks great @tofumatt – just the one question for you 🙂

File: assets/js/modules/pagespeed-insights/components/common/LabReportMetrics.js (hunk 3 above)
Comment (tofumatt) [reply assets/js/modules/pagespeed-insights/components/common/LabReportMetrics.js:131]: Oh, sweet, done!

**Review by aaemnnosttv — APPROVED** [review]
Super 👍
